pkg/manifests/manifests.go

```diff
@@ -290,6 +290,10 @@ func (f *Factory) AlertmanagerMain(host string) (*monv1.Alertmanager, error) {
 		a.Spec.NodeSelector = f.config.AlertmanagerMainConfig.NodeSelector
 	}
 
+	if f.config.AlertmanagerMainConfig.Tolerations != nil {
```

Comment:
best is to check `len(tolerations)` in each of these as in go a nil slice will also return length 0